### ISSUE TITLE
Protein inference - Protein FDR/PEP calculation

### ISSUE THREAD

**Issue by MiguelCos** [kusterlab/prosit#44]
Dear developers,

I have been happy with the rescoring results from Prosit, starting from a MaxQuant `msms.txt` file. I get > 2K target hits at the PSM level after rescoring.

I was wondering about what would you recommend for protein inference or protein FDR calculation, starting from the identified features after re-scoring.

Right now I am just using unique peptides to get the identified proteins but I would like to be able to assign an FDR value at the protein level.

Many thanks in advance.
Miguel

**Comment by LLautenbacher** [comment]
Hi Miguel,

@MatthewThe is currently working on implementing protein inference. I'm not sure how far along this is maybe he can give you some help on this.

Best,
Ludwig

**Comment by MiguelCos** [comment]
Hello Ludwig,

Many thanks for your comment. I have indeed exchanged some ideas with Matthew and managed to find a solution which I will post here if anyone else faces the same issue. The workflow is summarized in the flowchart below.

I made MQ generate the mzTab output which contains the complete mapping between peptide sequences and the proteins that they belong to (including the REV__ proteins generated by MQ). I modified the `.tab` file from Prosit to include both protein IDs and peptide sequences. Then I used an R script to correct the ScanNr annotation in the `.tab` file. I modified a fasta file, so the protein header would match the protein ID now found in the `.tab` file. Eliminated all features related to contaminants from the `.tab` file. And ran percolator using the modified fasta and the prosit-generated-modified `.tab` file.

Example command:

```
percolator data.tab -f data.fasta -P "REV__" -l prosit_target_proteins.tsv -L prosit_decoy_proteins.tsv
```
The decoy and target protein tsv files were generated nicely using this approach.

![image](https://user-images.githubusercontent.com/33185213/97421965-6f72dd00-190d-11eb-80ca-26ab55755417.png)

Best wishes,
Miguel
